__tests__/app.test.js

```diff
@@ -412,3 +412,90 @@ describe("POST /api/articles/:article_id/comments", () => {
+          created_at: "2020-10-16T05:03:00.000Z",
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+          votes: 3,
+        });
+      });
+  });
+  it("200: should return the selected article with the correct new number of votes - where article already has votes", () => {
+    const voteObj = { inc_votes: 3 };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(voteObj)
+      .expect(200)
+      .then((data) => {
+        expect(data.body.article).toMatchObject(  {
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 103,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
+  });
+  it('404: should respond with a 404 error if the article id is well formed but non-existant', () => {
+    const voteObj = {inc_votes: 3};
+    return request(app)
+    .patch("/api/articles/10000")
+    .send(voteObj)
+    .expect(404)
+    .then((data)=>{
+      expect(data.body.msg).toBe("Article not found")
+    })
+    
+  });
+  it('400: should respond with a 400 error if the article id is not well formed', () => {
+    const voteObj = {inc_votes: 3};
+    return request(app)
+    .patch("/api/articles/clogs")
+    .send(voteObj)
+    .expect(400)
+    .then((data)=>{
+      expect(data.body.msg).toBe("Bad request")
+    })
+  });
+
+  it('400: should respond with a 400 error if the request body contains an invalid key', () => {
+    const voteObj = {inc_votez: 3};
+    return request(app)
+    .patch("/api/articles/1")
+    .send(voteObj)
+    .expect(400)
+    .then((data)=>{
+      expect(data.body.msg).toBe("Bad request")
+    })
+  });
+  it('400: should respond with a 400 error if the request body contains an invalid value', () => {
+    const voteObj = {inc_votes: "clogs"};
+    return request(app)
+    .patch("/api/articles/1")
+    .send(voteObj)
+    .expect(400)
+    .then((data)=>{
+      expect(data.body.msg).toBe("Bad request")
+    })
+  });
```

Comment:
We can add a test with status 200 when the voteObj has additional properties that are ignored 🙂